### PR TITLE
Small batch of Clang tidy fixes

### PR DESCRIPTION
**Pull request by simonsan**

### CODE REVIEW

File: nyan/api_error.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ MemberTypeError::MemberTypeError(const fqon_t &objname, const memberid_t &member
                                  const std::string &real_type, const std::string &wrong_type)
 	:
 	APIError{
-	(static_cast<const std::ostringstream&>(
+	(dynamic_cast<const std::ostringstream&>(
```
Comment (TheJJ) [inline nyan/api_error.cpp]: static is intentional: we know it is a ostringstream just fine.

File: nyan/error.cpp [hunk 2]
```diff
@@ -85,13 +86,13 @@ constexpr const char *runtime_error_message = "polymorphic nyan error, catch by
 bool Error::break_on_error = false;
 
 
-Error::Error(const std::string &msg,
+Error::Error(std::string msg,
```
Comment (TheJJ) [inline nyan/error.cpp]: please use a `&&`

File: nyan/lang_error.cpp [hunk 3]
```diff
@@ -89,10 +90,10 @@ TypeError::TypeError(const Location &location, const std::string &msg)
 
 NameError::NameError(const Location &location,
                      const std::string &msg,
-                     const std::string &name)
+                     std::string name)
```
Comment (TheJJ) [inline nyan/lang_error.cpp]: `&&` pls

File: nyan/lang_error.h [hunk 4]
```diff
@@ -45,7 +45,7 @@ class TypeError : public LangError {
 class NameError : public LangError {
 public:
 	NameError(const Location &location,
-	          const std::string &msg, const std::string &name="");
+	          const std::string &msg, std::string name="");
```
Comment (TheJJ) [inline nyan/lang_error.h]: `&&` pls

File: nyan/member.cpp [hunk 5]
```diff
@@ -41,8 +41,8 @@ Member &Member::operator =(const Member &other) {
 
 
 Member &Member::operator =(Member &&other) noexcept {
-	this->override_depth = std::move(other.override_depth);
-	this->operation = std::move(other.operation);
+	this->override_depth = other.override_depth;
+	this->operation = other.operation;
```
Comment (TheJJ) [inline nyan/member.cpp]: move is intentional

File: nyan/location.h [hunk 6]
```diff
@@ -47,9 +47,9 @@ class Location {
 
 	std::shared_ptr<File> file;
 
-	int line;
-	int line_offset;
-	int length;
+	int line{};
+	int line_offset{};
+	int length{};
```
Comment (TheJJ) [inline nyan/location.h]: nope pls don't

**Comment by simonsan** [comment]
As my intention with this PR was also to probably find the error that was responsible for the win-build break, I changed the PR now to the immediate fixes that don't take any further work. All performance optimisation etc, we can do later, too.

**Pull request merged**